src/services/matrix.rs

```diff
@@ -72,47 +77,23 @@ pub mod matrix {
         pub send_to_thread: bool,
     }
 
-    pub async fn create_client(homeserver_url_str: String) -> Client {
+    pub async fn create_client(homeserver_url_str: &str) -> Client {
         info!("create client ");
         let homeserver_url =
             Url::parse(&homeserver_url_str).expect("Couldn't parse the homeserver URL");
-        let client = Client::new(homeserver_url).await.unwrap();
+        let client = Client::new(homeserver_url)
+            .await
+            .expect("can't handle new Client: create_client");
```

Comment:
This `create_client` function should return a Result that is later handled, the url being correct or the client always working are not things we can "expect" to happen
.